lib/model/CTokenListDataCategorizerBase.cc

```diff
@@ -539,6 +528,47 @@ std::size_t CTokenListDataCategorizerBase::memoryUsage() const {
     return mem;
 }
 
+void CTokenListDataCategorizerBase::updateMemoryResults(CResourceMonitor::SResults& results) const {
```

Comment:
I'm not sure about calling this `updateMemoryResults`. This includes other information now. Maybe `updateTelemetry`? I guess there is a slight issue because we've subverted `CResourceMonitor` to do more than monitor resource usage. Perhaps we should think about renaming this in a separate PR.

Reply:
CLion makes these renamings pretty trivial so how about `updateModelSizeStats`?  Then naming will be consistent through C++, Java and JavaScript, which might make it easier to trace how these numbers flow through the code.

Many of the new stats _are_ related to resource use - certainly `total_category_count`, `rare_category_count` and `dead_category_count` are indicators of how much memory categorization is using, and whether that usage is suboptimal.  To put the other stats somewhere else would just create complexity.

Reply:
Ok cool. I definitely think that's better.
> To put the other stats somewhere else would just create complexity.

Btw, I completely agree we don't want these somewhere else. It's more I wonder if we should think about renaming `CResourceMonitor` at some point. I'm happy for that not to happen in this PR though.

Reply:
++ to not renaming `CResourceMonitor` in this PR.  That would touch a very large number of files unrelated to categorization.